src/Xamarin.Android.Build.Tasks/Tasks/ResolveSdksTask.cs

```diff
@@ -92,12 +96,38 @@ public override bool Execute ()
 				return false;
 			}
 
+			try {
+				Log.LogDebugMessage ($"JavaSdkPath: {JavaSdkPath}");
+				Xamarin.Android.Tools.JdkInfo info = null;
+				try {
+					info = new Xamarin.Android.Tools.JdkInfo (JavaSdkPath);
+				} catch {
+					info = Xamarin.Android.Tools.JdkInfo.GetKnownSystemJdkInfos (this.CreateTaskLogger ()).FirstOrDefault ();
+				}
+
+				JdkJvmPath = info.JdkJvmPath;
+
+				if (string.IsNullOrEmpty (JdkJvmPath)) {
+					Log.LogCodedError ("XA5300", $"{nameof (JdkJvmPath)} is blank");
+					return false;
+				}
+
+				if (!File.Exists (JdkJvmPath)) {
+					Log.LogCodedError ("XA5300", $"JdkJvmPath not found at {JdkJvmPath}");
+					return false;
+				}
+			} catch (Exception e) {
+				Log.LogCodedError ("XA5300", $"Unable to find {nameof (JdkJvmPath)}{Environment.NewLine}{e}");
+				return false;
+			}
+
```

Comment:
@radekdoulik quick question about this task. There is a Visual Studio team very focused on the performance of "solution create", which is basically the initial design-time build from our targets.

Since looking up `JdkJvmPath` takes extra time, could it:
- Not run on `DesignTimeBuild=True`?
- Not run unless `AndroidGenerateJniMarshalMethods=True`?
- It also might be worth splitting this out into another MSBuild task?

Reply:
It is now only used by `jnimarshalmethod-gen`, so we can look it up only when `AndroidGenerateJniMarshalMethods` is *true*

Reply:
Ok, cool. I might look at making those changes ^^ I'll have you review.

Might not happen this week, though. 😃